### PR TITLE
fix: Fix table cell duplication and broken rowspan/colspan with footnotes

### LINKED ISSUE #1663: Table cell duplication when footnotes exist
Test case: [table-and-footnote.html](https://gist.githubusercontent.com/MurakamiShinyu/617e533a6d55bd4785a8a9b6be84aad7/raw/table-and-footnote.html)

Test with Vivliostyle Viewer v2.40.0:
https://vivliostyle.github.io/viewer/v2.40.0/#src=https://gist.githubusercontent.com/MurakamiShinyu/617e533a6d55bd4785a8a9b6be84aad7/raw/table-and-footnote.html

Screenshot:

Image

### LINKED ISSUE #1667: Broken table with rowspan/colspan and footnotes
Test sample: [table-rowspan-colspan-2.html](https://gist.github.com/MurakamiShinyu/617e533a6d55bd4785a8a9b6be84aad7#file-footnotes-in-table-rowspan-colspan-2-html)

Test with Vivliostyle Viewer (canary):
https://vivliostyle.vercel.app/#src=https://gist.github.com/MurakamiShinyu/617e533a6d55bd4785a8a9b6be84aad7#file-footnotes-in-table-rowspan-colspan-2-html

Screenshot:

Image

Problems:
- Cell D, which should continue from the previous page, is displayed from the beginning on Page 2. It's also positioned starting from Col1, not the original Col2. The footnote for FOOTNOTE4, which has a footnote reference in cell D, is also output on both Page 1 and Page 2.
- In the footnotes on Page 2, FOOTNOTE6 is above FOOTNOTE5, so the order is strange.
- Cell E is partially displayed on Page 2, and the content that should continue on Page 3 has disappeared.
- Cell G has disappeared.

### PR DESCRIPTION
**Pull request by MurakamiShinyu**
The fix in PR #1665 for table cell duplication with footnotes (issue #1663) was incomplete. When footnotes caused layout retries, table cells could still be duplicated or lost in certain cases. This commit addresses the remaining problems:

- Filter-based clearing in finishBreak to preserve `cellBreakPositions` entries from other rows instead of clearing all entries
- Use `.find()` by cell reference in `registerCellFragmentIndex` to avoid index mismatch after entries are reordered
- Add `hasUnfinishedCells()` check in `BetweenTableRowBreakPosition` to prevent accepting a between-row break when non-spanning cells have remaining content after footnote-triggered space reduction
- Save and restore `FormattingContext` states (`cellBreakPositions`) on column-level and page-level retries in `ops.ts` to prevent state loss

Additionally, fix broken table layout with rowspan/colspan and footnotes (issue #1667): `restoreState()` now creates a fresh copy of the `cellBreakPositions` array to prevent the saved state from being mutated when `extractRowSpanningCellBreakPositions` splices entries during subsequent layout attempts.

- Closes #1663
- Closes #1667

Assisted-by: GitHub Copilot Chat:gpt-5.3-codex

<details>
<summary>How the AI was used</summary>

- The human author reported that PR #1665's earlier fix for issue #1663 was incomplete (found via a page-size variation, not a regression) and asked for a re-fix.
- The human author iteratively found and reported further edge cases (a case where cell content still went missing, then a rowspan/colspan misalignment matching issue #1667) and decided which of the newly added test cases to keep in `file-list.js`, discarding redundant variants and renaming the one covering the rowspan/colspan case.
- The human author directed the commit message to cover both issue #1663 (incomplete fix) and #1667, created the PR, and asked the AI to check and respond to review comments.

</details>

### CODE REVIEW

File: packages/core/src/vivliostyle/ops.ts [hunk 1]
```diff
@@ -233,6 +233,60 @@ export class Style {
...
+      if (fcp.chunkPosition.floats) {
+        for (const floatPos of fcp.chunkPosition.floats) {
+          for (const step of floatPos.steps) {
+            if (step.formattingContext && !states.has(step.formattingContext)) {
+              states.set(
+                step.formattingContext,
+                step.formattingContext.saveState(),
+              );
+            }
+          }
```
[the remark this answers is not in the document]
Comment (MurakamiShinyu) [reply packages/core/src/vivliostyle/ops.ts:267]: Good catch. shadowSibling chains (e.g., footnote-call → footnote element) do carry their own formattingContext, which could theoretically be a different instance. In practice, they reference the same TableFormattingContext object as the parent steps (since both live in the same table), so the Map deduplication handles it. However, for correctness and to match the function's stated intent, I've updated collectFormattingContextStates to recursively traverse shadowSibling chains. Fixed in 249e9dc.
